xcube_geodb/core/user.py

```diff
@@ -0,0 +1,318 @@
+        return r
+
+    def get(self, path: str, params: Optional[Dict] = None):
+        """
+        Send a reload request to the GeoServer and clear the cache.
+        :return: The response given by the GeoServer.
+        """
+        get_url = urljoin(self.service_url, path)
+
+        r = self.session.get(get_url, params=params)
+        r.raise_for_status()
+
+        self._cache.clear()
+        return r
+
+    def reset(self):
+        """
+        Send a reset request to the GeoServer and clear the cache.
+        :return: The response given by the server.
+        """
+        reset_url = urljoin(self.service_url, "reset")
+        r = self.session.post(reset_url)
+        self._cache.clear()
+        return r
+
+    def post(self, path: str, payload: Dict, params: Optional[Dict] = None, content_type: str = "application/json"):
+        """
+
+        Args:
+            path:
+            payload:
+            params:
+            content_type:
+
+        Returns:
+
+        """
+        return self._save(self.session.post, path, payload, params, content_type)
+
+    def put(self, path: str, payload: Dict, params: Optional[Dict] = None, content_type: str = "application/json"):
+        """
+
+        Args:
+            path:
+            payload:
+            params:
+            content_type:
+
+        Returns:
+
+        """
+        return self._save(self.session.put, path, payload, params, content_type)
+
+    def _save(self, method, path: str, payload: Dict, params: Optional[Dict] = None,
+              content_type: str = "application/json"):
+        """
+        saves an object to the REST service
+        gets the object's REST location and the data from the object,
+        then POSTS the request.
+        :return: The response given by the server.
+
+        Args:
+            content_type (str):
+            obj (GeoserverObject):
+        """
+
+        rest_url = urljoin(self.service_url, path)
+
+        headers = {
+            "Content-type": content_type,
+            "Accept": content_type
+        }
+
+        r = method(rest_url, data=payload, params=params, headers=headers)
+        r.raise_for_status()
+
+        self._cache.clear()
+
+        return r
+
+
+class GeoServerObject:
+    def __init__(self, session: GeoserverSession = None):
+        if session is not None:
+            self._session = session
+        else:
+            self._session = GeoserverSession()
+
+    @abstractmethod
+    @property
+    def path(self):
+        pass
+
+
+class User(GeoServerObject):
+    path = "/rest/usergroup/users"
+
+    def __init__(self,
+                 user_name: str = GEOSERVER_DEFAULT_PARAMETERS['user_name'],
+                 password: str = GEOSERVER_DEFAULT_PARAMETERS['password'],
+                 active: bool = True,
+                 session: Optional[GeoserverSession] = None
+                 ):
+        super().__init__(session=session)
+
+        self._username = user_name
```

Comment:
1. `user_name`
2. `_username`

Reply:
The user is not used at this stage. I have removed it from this version but applied your changes for the next version where we are going to introduce a use class.